pkg/parse/parse.go

```diff
@@ -48,6 +49,22 @@ const (
 	BuildahCacheDir = "buildah-cache"
 )
 
+// RepoNameToNamedReference parse the raw string to Named reference
+func RepoNameToNamedReference(dest string) (reference.Named, error) {
+	ref, err := reference.Parse(dest)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse reference: %w", err)
+	}
+	named, ok := ref.(reference.Named)
```

Comment:
I think this should just be a `reference.ParseNormalizedNamed`. Otherwise _something_ is going to need to deal with the `docker.io[/library]` normalization, notably `c/image/docker.NewReference` expects a normalized value, and using raw `busybox` like the above would break it.

Reply:
`ParseNormalizedNamed(` accepts tag also afaics, it does not accepts string like `localhost/repo`, without tag it fails with invalid reference format hence i decided not to use it but let me know if something else can be used.

Reply:
> `ParseNormalizedNamed(` accepts tag also afaics

Sure, I wasn’t suggesting dropping the `IsNameOnly` check.

>, it does not accepts string like `localhost/repo`

I have just tested that it does accept exactly that string.

Reply:
On an edited unit test, showing how various inputs are normalized (input => `ref.String()`):
```
    normalize_test.go:58: "docker/docker" => "docker.io/docker/docker"
    normalize_test.go:58: "library/debian" => "docker.io/library/debian"
    normalize_test.go:58: "debian" => "docker.io/library/debian"
    normalize_test.go:58: "docker.io/docker/docker" => "docker.io/docker/docker"
    normalize_test.go:58: "docker.io/library/debian" => "docker.io/library/debian"
    normalize_test.go:58: "docker.io/debian" => "docker.io/library/debian"
    normalize_test.go:58: "index.docker.io/docker/docker" => "docker.io/docker/docker"
    normalize_test.go:58: "localhost/repo:tag" => "localhost/repo:tag"
    normalize_test.go:58: "localhost" => "docker.io/library/localhost"
    normalize_test.go:58: "notlocalhost" => "docker.io/library/notlocalhost"
    normalize_test.go:58: "index.docker.io/library/debian" => "docker.io/library/debian"
    normalize_test.go:58: "index.docker.io/debian" => "docker.io/library/debian"
    normalize_test.go:58: "127.0.0.1:5000/docker/docker" => "127.0.0.1:5000/docker/docker"
    normalize_test.go:58: "127.0.0.1:5000/library/debian" => "127.0.0.1:5000/library/debian"
    normalize_test.go:58: "127.0.0.1:5000/debian" => "127.0.0.1:5000/debian"
    normalize_test.go:58: "thisisthesongthatneverendsitgoesonandonandonthisisthesongthatnev" => "docker.io/library/thisisthesongthatneverendsitgoesonandonandonthisisthesongthatnev"
    normalize_test.go:58: "docker.io/1a3f5e7d9c1b3a5f7e9d1c3b5a7f9e1d3c5b7a9f1e3d5d7c9b1a3f5e7d9c1b3a" => "docker.io/library/1a3f5e7d9c1b3a5f7e9d1c3b5a7f9e1d3c5b7a9f1e3d5d7c9b1a3f5e7d9c1b3a"
```

Reply:
hmm it works for me now, I guess i just used the wrong function last time. Thanks I have amended the PR.